appservice/src/createAppService/AppInsightsLocationStep.ts

```diff
@@ -0,0 +1,105 @@
+    }
+
+    public shouldPrompt(wizardContext: IAppServiceWizardContext): boolean {
+        return !wizardContext.appInsightsComponent && !wizardContext.newAppInsightsLocation;
+    }
+
+    public async getSupportedLocation(wizardContext: IAppServiceWizardContext, location: Location): Promise<string | undefined> {
+        const resourceClient: ResourceManagementClient = createAzureClient(wizardContext, ResourceManagementClient);
+        const supportedRegions: Provider = await resourceClient.providers.get('microsoft.insights');
+        const componentsResourceType: ProviderResourceType | undefined = supportedRegions.resourceTypes && supportedRegions.resourceTypes.find(aiRt => aiRt.resourceType === 'components');
+        const locationName: string = nonNullProp(location, 'name');
+        // still need to do a name step
+        if (!!componentsResourceType && !!componentsResourceType.locations && componentsResourceType.locations.some((loc) => loc === location.displayName)) {
+            wizardContext.telemetry.properties.locationSupported = 'true';
+            return locationName;
+        } else {
+            // If there is no exact match, then query the regionMapping.json
+            const pairedRegions: string[] | undefined = await this.getPairedRegions(locationName);
+            if (pairedRegions.length > 0) {
+                // if there is at least one region listed, return the first
+                wizardContext.telemetry.properties.locationSupported = 'pairedRegion';
+                return pairedRegions[0];
+            }
+
+            wizardContext.telemetry.properties.locationSupported = 'false';
+            return undefined;
+        }
+    }
+
+    public async getPairedRegions(locationName: string): Promise<string[]> {
+        try {
+            const request: requestUtils.Request = await requestUtils.getDefaultRequest('https://appinsights.azureedge.net/portal/regionMapping.json');
+            const response: string = await requestUtils.sendRequest(request);
+            const regionMappingJson: RegionMappingJsonResponse = <RegionMappingJsonResponse>JSON.parse(response);
+
+            // tslint:disable-next-line: strict-boolean-expressions
+            if (regionMappingJson.regions[locationName]) {
+                return regionMappingJson.regions[locationName].pairedRegions;
+            }
+        } catch (error) {
+            // ignore the error
+        }
+        return [];
+    }
+
+    private async getQuickPicks(wizardContext: IAppServiceWizardContext): Promise<IAzureQuickPickItem<string | undefined>[]> {
+        let picks: IAzureQuickPickItem<string | undefined>[] = [];
+        const locationName: string = nonNullProp(nonNullProp(wizardContext, 'location'), 'name');
+        let pairedRegions: string[] = [locationName];
+        pairedRegions = pairedRegions.concat(await this.getPairedRegions(locationName));
+
+        const locations: Location[] = await LocationListStep.getLocations(wizardContext);
```

Comment:
Are you displaying the locations that specifically support app insights or all locations that support function apps?